### ISSUE TITLE
Add categories to the "New Page" pattern modal

### ISSUE THREAD

**Issue by simison** [WordPress/gutenberg#46498]
## What problem does this address?

[As implemented previously](https://github.com/WordPress/gutenberg/pull/40034), when registering patterns with  `'blockTypes' => array( 'core/post-content' )` annotation, they appear in a modal when creating a new page:

<img width="1303" alt="Screenshot 2022-12-13 at 12 41 20" src="https://user-images.githubusercontent.com/87168/207297323-d389f920-9961-4cf3-817c-a72d9be388c2.png">

The small modal becomes unwieldly very quickly when a theme or plugin registers dozens of patterns:

https://user-images.githubusercontent.com/87168/207298418-d4b4e7ae-bddb-45da-bf0a-f316f9e56866.mov

## What is your proposed solution?

- Add possibility to add categories.
- Increase the modal size to (almost) full browser size to accommodate categories and, be able to see patterns better
- Add search input. When it makes sense to add lots of patterns, it makes sense to let search them, too.

Basically re-use already existing "Explore all patterns" modal. That seemed to be intention [in the original issue](https://github.com/WordPress/gutenberg/issues/38787) as well.

<img width="1697" alt="Screenshot 2022-12-13 at 12 59 27" src="https://user-images.githubusercontent.com/87168/207300215-171e7852-0eca-4c5e-add1-b29345389d54.png">

Categories could be pulled from existing `categories` key, without needing to introduce new APIs.

**Comment by annezazu** [comment]
Tagging in @WordPress/gutenberg-design  on this one too :)

**Comment by paaljoachim** [comment]
Thank you for sharing @simison and @annezazu !
Sounds like a very good idea to me!
As a bigger modal will give more space to the explorations of the current patterns being looked at.

**Comment by jasmussen** [comment]
Proposed solution of using existing "Explore all" sounds good. 

I think there's still an opportunity for further unification at a later time. We have a couple of very adjacent modals:

* This one
* Explore all patterns
* Choose a starting pattern when adding a new page
* Choose a starting pattern when creating a new template

All of those could be considered as part of a whole, and look and behave similarly.

**Comment by jameskoster** [comment]
Re unification, I just wanted to point to issue #39308 which has [some designs](https://github.com/WordPress/gutenberg/issues/39308#issuecomment-1133086932) for that. They probably need some polishing, but I'm sharing here in case we want to implement any of those ideas.

**Comment by paaljoachim** [comment]
There is a PR here which addresses the modal size and is associated to this issue:
https://github.com/WordPress/gutenberg/pull/49722

**Comment by simison** [comment]
@jameskoster  @jasmussen I was looking at new page modal and while it looks great, it still feels like category-list would be very helpful for sites with lots of patterns?

![image](https://github.com/WordPress/gutenberg/assets/87168/e83bc8c7-fa3f-4661-af0f-719fce7f2e0e)

**Comment by jameskoster** [comment]
I'm wondering whether this modal could be updated to utilise the data view work and make use of all the associated features (search/filtering, view options, pagination, etc). That would allow users to filter by category, and anything else deemed useful.

**Comment by simison** [comment]
@scruffian I think this can be closed now that https://github.com/WordPress/gutenberg/pull/66836 merged?

**Comment by ntsekouras** [comment]
Yeah, let's close this one since we landed https://github.com/WordPress/gutenberg/pull/66836.
